OmniKit/MessageTransport/MessageBlocks/DetailedStatus.swift

```diff
@@ -34,9 +34,8 @@ public struct DetailedStatus : PodInfo, Equatable {
     public let data: Data
     
     public init(encodedData: Data) throws {
-        
-        guard let podInfoType = PodInfoResponseSubType(rawValue: encodedData[0]), podInfoType == self.podInfoType else {
-            throw MessageError.unknownValue(value: encodedData[0], typeDescription: "detailedStatus")
+        guard encodedData[0] == self.podInfoType.rawValue else {
+            throw MessageError.unknownValue(value: encodedData[0], typeDescription: "PodInfoResponseSubType")
```

Comment:
This is no longer an unknown value for PodInfoResponseSubType, so I don't think unknownValue is appropriate here.  The error is that you're expecting a podInfoType for [0] and getting something else. I don't see an appropriate MessageError type for this; might need to make a new one.  But as I stated before, I think you can just remove this check, no? We only get into this code if we've already checked that byte.

Same for the similar blocks below.

Reply:
Doing an assertionFailure() would be another way of keeping an assertion here for safety during development.

Reply:
The case I wanted to protect against where the podInfoType could be wrong would be a downcast to the wrong podInfo type such as in:
```quote
    let podInfoResponse: PodInfoResponse = try send([GetStatusCommand(podInfoType: .pulseLogRecent)])
    if let podInfoPulseLogPrevious = podInfoResponse.podInfo as? PodInfoPulseLogPrevious
```
I think using asserts would be the right way to go for this stuff, but I didn’t see them used that often in Loop and so I thought that its here might be frowned on.

Reply:
At the point of the downcast on this line, we have already checked that 1) the podInfoResponseSubType is a valid number, and 2) that the message has decoded successfully as that type.  That happens in the `public init(encodedData: Data)` constructor for PodInfoResponse. So at this point in the code, no matter what is returned from the pod, the parsing will not call a constructor for one of the PodInfo subtypes with the wrong value in byte[0].  The downcast will succeed or result in a null depending on whether the subtype was a PulseLogPrevious type.

Reply:
Got it. The self.podInfoType checking was only verifying that these values were initialized correctly in the source files. I have now removed these checks.